### ISSUE TITLE
Add E2E tests for testing library mode builds for mobile

### ISSUE THREAD

**Issue by ivanpovazan** [dotnet/sdk#42064]
### Description

Mono and NativeAOT support building custom managed libraries for mobile platforms that do not rely on target OS specific framework. This is achieved by publishing the class library by using the general "non-OS-specific" target framework i.e. `net8.0` and the runtime identifier for a specific mobile platform e.g.: `dotnet publish -r ios-arm64`. For a given class library the toolchain will produce `libClassLib.dylib` which can be then referenced from the mobile application (more info can be found here: https://github.com/dotnet/runtime/blob/5b5d7919a6099c9ea0056eaa800d59f53b3d9fe1/docs/design/mono/mono-library-mode.md).

dotnet/runtime repo includes functional tests which are testing library mode support for mobile, but these tests do not test the actual end-to-end scenario. 
To improve our test coverage and catch early any potential issues with this support, we should add appropriate test cases in this repository.

/cc: @steveisok

**Comment by Youssef1313** [comment]
I don't think this is related to `dotnet test`. Adding `area-runtime` as a best guess.
